src/app/shared-user/user/user.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { Router } from '@angular/router';
+import { UserStructure } from 'src/app/user-structure.model';
+
+@Component({
+  selector: 'app-user',
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+export class UserComponent {
+
+  className: string;
+
+  @Input() user: UserStructure;
+
+  @Output() changeStatus: EventEmitter<string> = new EventEmitter();
+
+  constructor(private route: Router){
+    if(this.route.url === '/active'){
+      this.className = 'active';
+    }else if(this.route.url === '/deleted'){
+      this.className = 'deleted';
+    }else{
+      this.className = 'manage';
+    }
```

Comment:
Instead of checking for the routes, you can pass an input to the component something like `showUserDetails` which is a boolean. Based on that value you can customize the html by showing the `Show Details` button.

Currently this logic of showing the `Show Details` button only works for the `manage` route. Imagine if I would like to have this component show `Show Details` button on some other route, then you need to change the logic (.ts and .html files) here to accommodate the behavior. Also if later you change the route names, then also this logic would break.